org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -89,10 +92,19 @@ public SignatureHelp signatureHelp(SignatureHelpParams position, IProgressMonito
 					int pos = offset;
 					if (method != null) {
 						int start = node.getStartPosition();
+						String nodeName = unit.getSource().substring(node.getStartPosition(), node.getStartPosition() + node.getLength());
+						int i = nodeName.lastIndexOf(".");
+						if (nodeName.startsWith("new ")) {
+							start = start + 4;
+						}
+						if (i > -1) {
+							start = start + i + 1;
+						}

```

Comment:
Could :

`((SimpleType)((ClassInstanceCreation)node).getType()).getOffset()` (with proper cast checking)

be used instead or relying on the String manipulation ?

Reply:
Fixed.